examples/antsforaging/antsforaging_ui/main.rs

```diff
@@ -0,0 +1,46 @@
+use crate::model::ant::Ant;
+use crate::visualization::custom_system_bundle::CustomSystemBundle;
+use crate::visualization::vis_state::VisState;
+use amethyst::renderer::palette::Srgba;
+use amethyst::utils::application_root_dir;
+use rust_ab::visualization::visualization::Visualization;
+use rust_ab::visualization::visualization_state::VisualizationState;
+
+pub mod model;
+pub mod visualization;
+
+pub const WIDTH: i64 = 200;
+pub const HEIGHT: i64 = 200;
+pub const NUM_AGENT: u128 = 500;
+pub const EVAPORATION: f64 = 0.999;
+pub const STEP: u128 = 50000;
+// Nest coordinate range
+pub const HOME_XMIN: i64 = 175;
+pub const HOME_XMAX: i64 = 175;
+pub const HOME_YMIN: i64 = 175;
+pub const HOME_YMAX: i64 = 175;
+// Food coordinate range
+pub const FOOD_XMIN: i64 = 25;
+pub const FOOD_XMAX: i64 = 25;
+pub const FOOD_YMIN: i64 = 25;
+pub const FOOD_YMAX: i64 = 25;
+
+fn main() -> amethyst::Result<()> {
+    let app_root = application_root_dir()?
+        .join("examples")
+        .join("antsforaging")
+        .join("antsforaging_ui");
+    /* CLion debugger starts the built executable directly from the target folder, so the path must
+            be hardcoded like this to be able to debug simulations.
+    let app_root = PathBuf::from(
+        "C:\\Users\\frafo\\Desktop\\Tirocinio\\abm\\examples\\antsforaging\\antsforaging_ui",
```

Comment:
Non puoi mettere un path assoluto? Il file appartiene al progetto

Reply:
Intendi un path relativo? Si, tenendo conto che l'esecuzione del debugger di CLion consiste nell'eseguire l'exe dalla cartella target, posso navigare in su fino alla root del progetto per poi scendere nella cartella dell'esempio, ho voluto semplicemente lasciare il mio commento come fix temporaneo nel caso in futuro si abbia la necessità di usare il debugger per questi esempi. Durante l'esecuzione "standard" di un esempio, viene usato `cargo run` che setta automaticamente la variabile d'ambiente `CARGO_MANIFEST_DIR` alla root del progetto, quindi in quel caso questo problema non è presente. Se l'eseguibile viene lanciato manualmente, l'app root di default é settata uguale alla cwd, infatti nel caso di CLion la cwd è la cartella target. Stavo pensando di fare un readme dedicato al framework di visualizzazione dove vengono esposte le funzionalità, l'architettura e gli accorgimenti da seguire per l'esecuzione corretta di un modello, come appunto usare preferibilmente cargo run, o nel caso di modelli aventi RustAB come dipendenza cargo, lanciare l'eseguibile con cwd contenente le cartelle assets e config (usate da Amethyst)

Reply:
Si intendevo il path relativo. In ogni caso sicuramente fare un readme apposito è sicuramente meglio, ti dovrebbe venire anche semplice visti tutti i commenti che hai all interno del codice che sono molto esplicativi.
Ad ogni modo meglio eliminare almeno la parte relativa al percorso tuo personale del pc, anche per avere un codice più corretto e leggibile